src/vs/editor/common/controller/cursorTypeOperations.ts

```diff
@@ -194,40 +219,53 @@ export class TypeOperations {
 		}
 	}
 
+
 	private static _enter(config: CursorConfiguration, model: ITokenizedModel, keepPosition: boolean, range: Range): EditOperationResult {
+		let enterAction = LanguageConfigurationRegistry.getEnterActionAtPosition(model, range.startLineNumber, range.startColumn);
+		let lineText = model.getLineContent(range.startLineNumber);
+		let indentation = strings.getLeadingWhitespace(lineText);
+		if (indentation.length > range.startColumn - 1) {
+			indentation = indentation.substring(0, range.startColumn - 1);
+		}
 
-		let r = LanguageConfigurationRegistry.getEnterActionAtPosition(model, range.startLineNumber, range.startColumn);
-		let enterAction = r.enterAction;
-		let indentation = r.indentation;
+		// adjust indentation of current line.
+		let beforeText = '';
+		if (enterAction.outdentCurrentLine) {
```

Comment:
When running scripts\test.bat --coverage this block is not covered, can you please write some tests that cover this new functionality?